include/solvers/cahn_hilliard_scratch_data.h

```diff
@@ -238,13 +265,58 @@ class CahnHilliardScratchData
               trace(this->hess_phi_potential[q][k]);
           }
       }
+
+    this->is_boundary_cell = cell->at_boundary();
+    if (cell->at_boundary())
```

Comment:
```suggestion
    if ( cell->at_boundary())
```
This would be sufficient no? Except if you reuse the variable below.

Reply:
One of the assemblers checks for boundary cells, so I need to update the is_boundary_attribute. However it is not necessary to use cell->at_boundary() again in the if, just use the attribute that we've just updated